ApplicationLibCode/ProjectDataModel/Flow/RimWellConnectivityTable.cpp

```diff
@@ -0,0 +1,1256 @@
+        }
+    }
+
+    return std::vector<QString>( productionWellNames.begin(), productionWellNames.end() );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+QString RimWellConnectivityTable::createTableTitle() const
+{
+    RiaDefines::EclipseUnitSystem     unitSet   = m_case->eclipseCaseData()->unitsType();
+    RimWellLogFile::WellFlowCondition condition = RimWellLogFile::WELL_FLOW_COND_RESERVOIR;
+
+    auto timeSampleValueTypeText = [&]() -> QString {
+        if ( m_timeSampleValueType == TimeSampleValueType::FLOW_RATE_PERCENTAGE )
+        {
+            return QString( "Percentage of Total Reservoir Flow Rate [%]" );
+        }
+        if ( m_timeSampleValueType == TimeSampleValueType::FLOW_RATE_FRACTION )
+        {
+            return QString( "Fraction of Total Reservoir Flow Rate" );
+        }
+        if ( m_timeSampleValueType == TimeSampleValueType::FLOW_RATE )
+        {
+            return "Total " + RimWellPlotTools::flowPlotAxisTitle( condition, unitSet );
+        }
+        return QString();
+    };
+
+    auto timeRangeValueTypeText = [&]() -> QString {
+        if ( m_timeRangeValueType() == TimeRangeValueType::ACCUMULATED_FLOW_VOLUME )
+        {
+            return "Accumulated Total " + RimWellPlotTools::flowVolumePlotAxisTitle( condition, unitSet );
+        }
+        if ( m_timeRangeValueType() == TimeRangeValueType::ACCUMULATED_FLOW_VOLUME_PERCENTAGE )
+        {
+            return QString( "Accumulated Total Reservoir Flow Volume Allocation [%]" );
+        }
+        if ( m_timeRangeValueType() == TimeRangeValueType::ACCUMULATED_FLOW_VOLUME_FRACTION )
+        {
+            return QString( "Accumulated Total Reservoir Flow Volume Allocation Fraction" );
+        }
+        return QString();
+    };
+
+    if ( m_timeStepSelection() == TimeStepSelection::SINGLE_TIME_STEP )
+    {
+        return QString( "%1 (%2)<br>Date: %3</br>" )
+            .arg( timeSampleValueTypeText() )
+            .arg( m_case->caseUserDescription() )
+            .arg( m_selectedTimeStep().toString( dateFormatString() ) );
+    }
+    if ( m_timeStepSelection() == TimeStepSelection::TIME_STEP_RANGE )
+    {
+        return QString( "%1 (%2)<br>Date range: %3 - %4, Number of time steps: %5</br>" )
+            .arg( timeRangeValueTypeText() )
+            .arg( m_case->caseUserDescription() )
+            .arg( m_selectedFromTimeStep().toString( dateFormatString() ) )
+            .arg( m_selectedToTimeStep().toString( dateFormatString() ) )
+            .arg( m_timeStepCount() );
+    }
+
+    return "";
+}
+
+std::pair<double, double> RimWellConnectivityTable::createLegendMinMaxValues( const double maxTableValue ) const
+{
+    if ( ( m_timeStepSelection == TimeStepSelection::SINGLE_TIME_STEP && m_timeSampleValueType == TimeSampleValueType::FLOW_RATE_PERCENTAGE ) ||
```

Comment:
Add `using enum TimeStepSelection;` to start of function, and remove `TimeStepSelection::` to improve readability